cmd/kueuectl/app/create/create_resourceflavor.go

```diff
@@ -0,0 +1,352 @@
+	var taints []corev1.Taint
+	uniqueTaints := make(map[corev1.TaintEffect]sets.Set[string])
+
+	for _, taintSpec := range spec {
+		newTaint, err := parseTaint(taintSpec)
+		if err != nil {
+			return nil, err
+		}
+		// validate if taint is unique by <key, effect>
+		if len(uniqueTaints[newTaint.Effect]) > 0 && uniqueTaints[newTaint.Effect].Has(newTaint.Key) {
+			return nil, fmt.Errorf("duplicated taints with the same key and effect: %v", newTaint)
+		}
+		// add taint to existingTaints for uniqueness check
+		if len(uniqueTaints[newTaint.Effect]) == 0 {
+			uniqueTaints[newTaint.Effect] = sets.New[string]()
+		}
+		uniqueTaints[newTaint.Effect].Insert(newTaint.Key)
+
+		taints = append(taints, newTaint)
+	}
+
+	return taints, nil
+}
+
+// parseTaint parses a taint from a string, whose form must be either
+// 'KEY=VALUE:EFFECT' or 'KEY:EFFECT'
+func parseTaint(str string) (corev1.Taint, error) {
+	var taint corev1.Taint
+
+	if len(str) == 0 {
+		return taint, errors.New("invalid taint spec")
+	}
+
+	var key string
+	var value string
+	var effect corev1.TaintEffect
+
+	parts := strings.Split(str, ":")
+	if len(parts) != 2 {
+		return taint, fmt.Errorf("invalid taint spec: %v", str)
+	}
+
+	effect = corev1.TaintEffect(parts[1])
+	if err := validateTaintEffect(effect); err != nil {
+		return taint, err
+	}
+
+	partsKV := strings.Split(parts[0], "=")
+	if len(partsKV) > 2 {
+		return taint, fmt.Errorf("invalid taint spec: %v", str)
+	}
+	key = partsKV[0]
+	if errs := validation.IsQualifiedName(key); len(errs) > 0 {
+		return taint, fmt.Errorf("invalid taint spec: %v, %s", str, strings.Join(errs, "; "))
+	}
+
+	if len(partsKV) == 2 {
+		value = partsKV[1]
+		if errs := validation.IsValidLabelValue(value); len(errs) > 0 {
+			return taint, fmt.Errorf("invalid taint spec: %v, %s", str, strings.Join(errs, "; "))
+		}
+	}
+
+	taint.Key = key
+	taint.Value = value
+	taint.Effect = effect
+
+	return taint, nil
+}
+
+func validateTaintEffect(effect corev1.TaintEffect) error {
+	if effect != corev1.TaintEffectNoSchedule && effect != corev1.TaintEffectPreferNoSchedule && effect != corev1.TaintEffectNoExecute {
+		return fmt.Errorf("invalid taint effect: %v, unsupported taint effect", effect)
+	}
+	return nil
+}
+
+// parseTolerations takes a spec which is an array and creates slices for new tolerations to be added.
+// It also validates the spec.
+func parseTolerations(spec []string) ([]corev1.Toleration, error) {
+	var tolerations []corev1.Toleration
+	uniqueTolerations := make(map[corev1.TaintEffect]sets.Set[string])
+
+	for _, tolerationSpec := range spec {
+		newTolerations, err := parseToleration(tolerationSpec)
+		if err != nil {
+			return nil, err
+		}
+		// validate if toleration is unique by <key, effect>
+		if len(uniqueTolerations[newTolerations.Effect]) > 0 && uniqueTolerations[newTolerations.Effect].Has(newTolerations.Key) {
+			return nil, fmt.Errorf("duplicated tolerations with the same key and effect: %v", newTolerations)
+		}
```

Comment:
should the value be relevant and operator?

Reply:
For tolerations we don't have this validation. So removed from cmd.